shell/platform/android/io/flutter/embedding/engine/dart/DartExecutor.java

```diff
@@ -0,0 +1,277 @@
+   * <p>
+   * See {@link DartCallback} for configuration options.
+   *
+   * @param dartCallback specifies which Dart callback to run, and where to find it
+   */
+  public void executeDartCallback(DartCallback dartCallback) {
+    if (isApplicationRunning) {
+      Log.w(TAG, "Attempted to run a DartExecutor that is already running.");
+      return;
+    }
+
+    flutterJNI.runBundleAndSnapshotFromLibrary(
+        new String[]{
+            dartCallback.pathToPrimaryBundle,
+            dartCallback.pathToFallbackBundle
+        },
+        dartCallback.callbackHandle.callbackName,
+        dartCallback.callbackHandle.callbackLibraryPath,
+        dartCallback.androidAssetManager
+    );
+
+    isApplicationRunning = true;
+  }
+
+  //------ START BinaryMessenger -----
+
+  /**
+   * Sends the given {@code message} from Android to Dart over the given {@code channel}.
+   *
+   * @param channel the name of the logical channel used for the message.
+   * @param message the message payload, a direct-allocated {@link ByteBuffer} with the message bytes
+   */
+  @Override
+  public void send(String channel, ByteBuffer message) {
+    messenger.send(channel, message, null);
+  }
+
+  /**
+   * Sends the given {@code messages} from Android to Dart over the given {@code channel} and
+   * then has the provided {@code callback} invoked when the Dart side responds.
+   *
+   * @param channel  the name of the logical channel used for the message.
+   * @param message  the message payload, a direct-allocated {@link ByteBuffer} with the message bytes
+   *                 between position zero and current position, or null.
+   * @param callback a callback invoked when the Dart application responds to the message
+   */
+  @Override
+  public void send(String channel, ByteBuffer message, BinaryMessenger.BinaryReply callback) {
+    messenger.send(channel, message, callback);
+  }
+
+  /**
+   * Sets the given {@link io.flutter.plugin.common.BinaryMessenger.BinaryMessageHandler} as the
+   * singular handler for all incoming messages received from the Dart side of this Dart execution
+   * context.
+   *
+   * @param channel the name of the channel.
+   * @param handler a {@link BinaryMessageHandler} to be invoked on incoming messages, or null.
+   */
+  @Override
+  public void setMessageHandler(String channel, BinaryMessenger.BinaryMessageHandler handler) {
+    messenger.setMessageHandler(channel, handler);
+  }
+  //------ END BinaryMessenger -----
+
+  /**
+   * Configuration options that specify which Dart entrypoint function is executed and where
+   * to find that entrypoint and other assets required for Dart execution.
+   */
+  public static class DartEntrypoint {
+    /**
+     * Standard Android AssetManager, provided from some {@code Context} or {@code Resources}.
+     */
+    public final AssetManager androidAssetManager;
+
+    /**
+     * The first place that Dart will look for a given function or asset.
```

Comment:
nit, optional: Not sure what formatter you're using, but short Javadoc can be fit into one line. Most formatters do this automatically.

`/** Example doc. */`

Reply:
Is there any particular reason to prefer that?  Right now all the properties are using the multiline version...and I actually find it easier to read with a little whitespace.

Reply:
No, it's totally subjective. It would matter if our format check complained about it, but it looks like everything's fine.